crates/re_viewport/src/context_menu.rs

```diff
@@ -0,0 +1,296 @@
+//TODO(ab): use list items to make those context menu nice to look at
+
+use crate::{Contents, ViewportBlueprint};
+use itertools::Itertools;
+use re_log_types::{EntityPath, EntityPathFilter};
+use re_space_view::{DataQueryBlueprint, SpaceViewBlueprint};
+use re_viewer_context::{ContainerId, Item, SpaceViewClassIdentifier, ViewerContext};
+
+/// Trait for things that can populate a context menu
+trait ContextMenuItem {
+    //TODO(ab): should probably return `egui::WidgetText` instead
+    fn label(&self, _ctx: &ViewerContext<'_>, _viewport_blueprint: &ViewportBlueprint) -> String {
```

Comment:
Just a label is a bit little though? Should add icons in a follow-up pr so we can show the familiar space view and layout icons - they are much faster to parse than text

Reply:
Yup—see my other comment.